ProcessLib/Process.cpp

```diff
@@ -34,7 +35,8 @@ Process::Process(
       _nonlinear_solver(nonlinear_solver),
       _time_discretization(std::move(time_discretization)),
       _convergence_criterion(std::move(convergence_criterion)),
-      _process_variables(std::move(process_variables))
+      _process_variables(std::move(process_variables)),
```

Comment:
Would be nice if the initializer list order is the same as parameter order. Suggestion: move `parameters` parameter after `convergence_criterion`.